src/node_os.cc

```diff
@@ -365,38 +366,31 @@ static void GetUserInfo(const FunctionCallbackInfo<Value>& args) {
                                                    pwd.username,
                                                    encoding,
                                                    &error);
+  if (username.IsEmpty()) {
+  fail_with_stringbytes_error:
+    uv_os_free_passwd(&pwd);
+    env->isolate()->ThrowException(error);
+    return;
+  }
```

Comment:
Good idea.  I was thinking of guarding the Encode() calls with an `error.IsEmpty()` check but that's a good idea.